src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -445,7 +445,14 @@ internal partial class BoundCatchBlock : ICatchClause
 
         IOperation ICatchClause.Filter => this.ExceptionFilterOpt;
 
-        ILocalSymbol ICatchClause.ExceptionLocal => this.LocalOpt;
+        ILocalSymbol ICatchClause.ExceptionLocal
+        {
+            get
+            {
+                var local = this.Locals.FirstOrDefault();
+                return local?.DeclarationKind == LocalDeclarationKind.CatchVariable ? local : null;
```

Comment:
> There is no need to do the search, if exception local is present, its the first one.

That conflicts with the comment in BoundNodes.xml:
```none
    In the initial bound tree the first variable is the exception variable (if present).
    After the node is lowered it might be another variable.
```

`IOperation` needs to work both pre-lowering and post-lowering.